### ISSUE TITLE
Оптимизация кадрового движение

### ISSUE THREAD

**Issue by gsbelarus** [GoldenSoftwareLtd/gedemin#2573]
Originally reported on Google Code with ID 2573

```
На больших базах много времени тратится на нахождение нужного кадрового движения. Рассмотрим
класс Twg_EmplMoveList(Twg_MoveCardList/Twg_MoveCard). Основная проблема там заключается
в том, что повсеместно используется ф-ция CreateMoveCardList. Она берёт всё движение
сотрудников за весь период и загоняет его словарь, и потом работает со словарём через
перебор.

Предлагается следующее:
1. Можно переписать все методы на простые запросы к БД на нужную дату без использования.
2. Можно переделать движение так, что бы в список попадало только нужное (например
за расчётный период) + учесть возможность добавления новых периодов с список.

```

Reported by `Alexander.GoldenSoft` on 2011-09-11 13:47:10

**Comment by gsbelarus** [comment]
```
Расчет итогового расчета замедляется, видимо, из-за вот этой проблемы. Поэтому есть
смысл посмотреть, что и как можно это реализовать. Так как на больших предприятиях
много тратится на расчет итогового начисления - 1-1,5 часа.
```

Reported by `roberto1813` on 2012-01-20 12:30:37

**Comment by gsbelarus** [comment]
Reported by `roberto1813` on 2014-03-17 12:44:01
- **Status changed**: `Duplicate`
- **Merged into**: #3332
